File: basics/구현/상하좌우.cpp
#include <string>

using namespace std;

int n;
string plans;
int x = 1, y = 1;

int dx[4] = {0, 0, -1, 1};
int dy[4] = {-1, 1, 0, 0};
char moves[4] = {'L', 'R', 'U', 'D'};

int main(){
  cin >> n;
  cin.ignore(); // 버퍼 비우기
  getline(cin, plans); // 이동 계획을 하나씩 확인

  for (int i = 0; i < plans.size(); i++){
    int nx, ny;
    for (int j = 0; j < 4; j++){
      if (plans[i] == moves[j]){
        nx = x + dx[j];
        ny = y + dy[j];
      }
    }
    if (nx < 1 || nx > n || ny < 1 || ny > n)
      continue;
    else
      x = nx;
      y = ny;
  }

  cout << x << ' ' << y << '\n';
  return 0;
}

File: basics/구현/상하좌우.py
n = int(input())
plans = input().split()

x, y = 1, 1
dx = [0, 0, -1, 1]
dy = [-1, 1, 0, 0]
moves = ['L', 'R', 'U', 'D'] # 왼쪽, 오른쪽, 위, 아래

for plan in plans:
    # 이동 후 좌표
    for i in range(len(moves)):
        if plan == moves[i]:
            nx, ny = x + dx[i], y + dy[i]
    # 벗어나는지 확인
    if nx < 1 or nx > n or ny < 1 or ny > n:
        continue
    else:
        x, y = nx, ny

print(x, y)
